pkg/package-server/storage/packagemanifest/watcher.go

```diff
@@ -0,0 +1,57 @@
+package packagemanifest
+
+import (
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/watch"
+
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/package-server/apis/packagemanifest/v1alpha1"
+)
+
+type PackageManifestWatcher struct {
+	namespace     string
+	labelSelector labels.Selector
+	stop          chan struct{}
+	results       chan watch.Event
+	in            <-chan v1alpha1.PackageManifest
+}
+
+func NewPackageManifestWatcher(namespace string, labelSelector labels.Selector) *PackageManifestWatcher {
+	watcher := &PackageManifestWatcher{
+		namespace:     namespace,
+		labelSelector: labelSelector,
+		stop:          make(chan struct{}),
+		results:       make(chan watch.Event),
+		in:            make(<-chan v1alpha1.PackageManifest),
+	}
+
+	return watcher
+}
+
+func (w *PackageManifestWatcher) Run() {
+	go func() {
+		for {
+			select {
+			case manifest := <-w.in:
```

Comment:
it would be nice if we could emit created/deleted here too, but I see that may be difficult. Other controllers can get that info from etcd events, but we don't have those here.